src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,147 @@
+ *         })(),
+ *     ]
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    private int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
+
+    /**
+     * @param mixed[] $structure Basic structure of the JSON containing replace identifiers
+     * @param array<string, iterable<mixed> $generators One or multiple generators indexed by replace identifier
+     * @param int                            $status    The HTTP status code (200 "OK" by default)
+     * @param array<string, string|string[]> $headers   An array of HTTP headers
+     * @param int                            $flushSize After every which item of a generator the flush function should be called
+     */
+    public function __construct(
+        private readonly array $structure,
+        private readonly array $generators,
+        int $status = 200,
+        array $headers = [],
+        private int $flushSize = 500,
+    ) {
+        parent::__construct($this->stream(...), $status, $headers);
+
+        if (!$this->headers->get('Content-Type')) {
+            $this->headers->set('Content-Type', 'application/json');
+        }
+    }
+
+    private function stream(): void
+    {
+        $keys = array_keys($this->generators);
+
+        $jsonEncodingOptions = \JSON_THROW_ON_ERROR | $this->getEncodingOptions();
+        $structureText = json_encode($this->structure, $jsonEncodingOptions);
+
+        foreach ($keys as $key) {
+            [$start, $end] = explode('"'.$key.'"', $structureText, 2);
+
+            // send first and between parts of the structure
+            echo $start;
+
+            $count = 0;
+            echo '[';
+            foreach ($this->generators[$key] as $array) {
+                if (0 !== $count) {
+                    // if not first element of the generic a separator is required between the elements
+                    echo ',';
+                }
+
+                echo json_encode($array, $jsonEncodingOptions);
+                ++$count;
+
+                if (0 === $count % $this->flushSize) {
+                    flush();
+                }
+            }
+            echo ']';
+
+            $structureText = $end;
+        }
+
+        echo $structureText; // send the after part of the structure as last
+    }
+
+    /**
+     * Returns options used while encoding data to JSON.
+     */
+    public function getEncodingOptions(): int
+    {
+        return $this->encodingOptions;
+    }
+
+    /**
+     * Sets options used while encoding data to JSON.
+     *
+     * @return $this
+     */
+    public function setEncodingOptions(int $encodingOptions): static
+    {
+        $this->encodingOptions = $encodingOptions;
+
+        return $this;
+    }
+
+    /**
+     * Sets the flush size.
+     */
+    public function getFlushSize(): int
+    {
+        return $this->flushSize;
+    }
+
+    /**
+     * Returns the flush size.
```

Comment:
Description is wrong

"Sets the flush size."

Reply:
Thx fixed the phpdocs